packages/flutter/lib/src/widgets/form.dart

```diff
@@ -417,7 +417,7 @@ class _FormScope extends InheritedWidget {
 /// otherwise.
 ///
 /// Used by [FormField.validator].
-typedef FormFieldValidator<T> = String? Function(T? value);
+typedef FormFieldValidator<T> = String? Function(BuildContext context, T? value);
```

Comment:
Unfortunately, changing the function signature this way qualifies as a [**breaking change**](https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes), since even if we update all the code within Flutter, any external code set up with the old function signature would suddenly break.

Reply:
I believe the ideal way to handle this would be to deprecate `FormFieldValidator` and introduce a new typedef with a different name.

We'd also need to add a [**data-driven fix**](https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md).

Reply:
Hello, thanks. I thought it was a breaking change but I didn't have an idea of other way to do it, but yes deprecating should be way better. Can you help me to define the flow to fix it? I have this in mind:
- Close this PR and open a new one
- Deprecate FormFieldValidator with corresponding decorator
- Adding the new typedef below `FormFieldValidator`, maybe naming it **FormFieldValidatorWithContext ??**
- Tests that use deprecate `FormFieldValidator` should be change to `FormFieldValidatorWithContext`??
- At the end I will see how to add a data-driven fix

Reply:
This looks like a good plan 👍

<br>

> Adding the new typedef below `FormFieldValidator`, maybe naming it **FormFieldValidatorWithContext ??**

Yeah, coming up with a good replacement name is going to be tricky.

<br>

> Tests that use deprecate `FormFieldValidator` should be change to `FormFieldValidatorWithContext`??

Yes, eventually, but it can probably happen after the old typedef is deprecated in a separate PR.